src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -298,19 +326,27 @@ private function extractFromReflectionType(\ReflectionType $reflectionType, \Ref
             $type = new Type(Type::BUILTIN_TYPE_NULL, $nullable);
         } elseif ($reflectionType->isBuiltin()) {
             $type = new Type($phpTypeOrClass, $nullable);
+        } elseif (null !== $reflectionMethod) {
+            $type = new Type(Type::BUILTIN_TYPE_OBJECT, $nullable, $this->resolveTypeName($phpTypeOrClass, $reflectionMethod, null));
+        } elseif (null !== $reflectionProperty) {
+            $type = new Type(Type::BUILTIN_TYPE_OBJECT, $nullable, $this->resolveTypeName($phpTypeOrClass, null, $reflectionProperty));
         } else {
-            $type = new Type(Type::BUILTIN_TYPE_OBJECT, $nullable, $this->resolveTypeName($phpTypeOrClass, $reflectionMethod));
+            throw new \BadMethodCallException('You should provide method or property reflection');
         }
 
         return $type;
     }
 
-    private function resolveTypeName(string $name, \ReflectionMethod $reflectionMethod): string
+    private function resolveTypeName(string $name, ?\ReflectionMethod $reflectionMethod, ?\ReflectionProperty $reflectionProperty): string
```

Comment:
as we only need `->getDeclaringClass()`, which is common on both class, I suggest using a single argument accepting `\ReflectionMethod|\ReflectionProperty` (and same in `extractFromReflectionType`). This will keep the code simpler

Reply:
Better than contribute with something is learn something contributing.

Reply:
Actually this doesn't seems to be a valid syntax. Is your suggestion something like this? 

```php
private function extractFromReflectionType(\ReflectionType $reflectionType, \ReflectionMethod|\ReflectionProperty $reflector): Type {...}
```

Or are you suggesting to not declare this argument with type:

```php
/**
 * @param \ReflectionMethod|\ReflectionProperty $reflector
*/
private function extractFromReflectionType(\ReflectionType $reflectionType, object $reflector): Type {...}
```

Reply:
both ReflectionMethod and \ReflectionProperty implements Reflector, so you can declare it like this:

```php
/**
 * @param \ReflectionMethod|\ReflectionProperty $reflector
*/
private function extractFromReflectionType(\ReflectionType $reflectionType, \Reflector $reflector): Type {...}
```

Reply:
Yes, both implements `\Reflector` but `\Reflector` doesn't have the necessary `getDeclaredClass()` method. In this case, I don't think that we should use this interface as a typehint.

Reply:
In any case It is far better than just `object`. For more specific typehint `@param` declaration will be enough.

Reply:
As both `extractFromReflectionType` and `resolveTypeName` are private methods, I'm good to remove any typehint for this argument and declare its type  on `@param` docblock. What do you think?